src/response/Response.cpp

```diff
@@ -0,0 +1,140 @@
+#include "../../include/Response.hpp"
+
+Response::Response() {}
+
+Response::Response(Request &request) {
+    this->_request = request;
+}
+
+Response::Response(Response const &src) {
+    *this = src;
+}
+
+Response::~Response() {}
+
+Response &Response::operator=(Response const &rhs) {
+    if (this != &rhs) {
+        this->_request = rhs._request;
+        this->_statusCode = rhs._statusCode;
+        this->_statusMessage = rhs._statusMessage;
+        this->_contentType = rhs._contentType;
+        this->_contentLength = rhs._contentLength;
+        this->_body = rhs._body;
+    }
+    return *this;
+}
+
+void Response::setRequest(Request &request) {
+    this->_request = request;
+}
+
+void Response::setStatusCode(int status_code) {
+    this->_statusCode = status_code;
+}
+
+void Response::setStatusMessage(std::string status_message) {
+    this->_statusMessage = status_message;
+}
+
+void Response::setContentType(std::string content_type) {
+    this->_contentType = content_type;
+}
+
+void Response::setContentLength(int content_length) {
+    this->_contentLength = content_length;
+}
+
+void Response::setBody(std::string body) {
+    this->_body = body;
+}
+
+void Response::createResponse() {
+    std::string response = "HTTP/1.1 " + std::to_string(this->_statusCode) + " " + this->_statusMessage + "\r\n";
+    response += "Content-Type: " + this->_contentType + "\r\n";
+    response += "Content-Length: " + std::to_string(this->_contentLength) + "\r\n";
+    response += "\r\n";
+    response += this->_body;
+    this->_response = response;
+}
+
+void Response::displayResponse() {
+    std::cout << this->_response << std::endl;
+}
+
+void Response::checkMethod(Request &request, Server &server, std::string location) {
+    std::string method = request.getRequestMethod();
+    std::vector<DirSettings> dirSettings = server.getDirSettings();
+    for (std::vector<DirSettings>::iterator it = dirSettings.begin(); it != dirSettings.end(); ++it) {
+        if (it->getLocation() == location) {
+            for (std::vector<std::string>::iterator it2 = it->getMethods().begin(); it2 != it->getMethods().end(); ++it2) {
+                if (*it2 == method) {
+                    sendResponse(method, location, request, it->getExecPath());
+                    return;
```

Comment:
Sorry, now Server class is being changed (now Liwen is modifying that with Config Parser parts, it's on reviewing), 

```
std::string			_serverName;
std::string			_rootDir;	// the place to store the files for this server (base directory to find the file)
DirSettings			_rootDirSettings;	// default setting
std::vector<DirSettings>	_optDirSettings;	// setting for special directories
std::vector<int>		_listPort;
std::vector<SockListen>		_listenSocks;
```
We have to check with "_optDirSettings" at first (this stores the special directories setting).
And if we can't find this location setting in _optDirSettings, we have to check the "_rootDirSettings".
"_rootDirSettings" is the default setting, and this setting effects to all directory which is not defined in optDirSettings.

Reply:
Hi, I added the checkMethod function in Request class, so the method check is already done before making response.
I reffered your code and added functions to get the corresponding Server and DirSettings in util.cpp, maybe it is useful for Response part also.

```
  Server	findServer(std::string name, std::vector<Server> list_server);
  DirSettings	findDirSetting(Request &request, Server &server);
  bool		checkMethodAllowed(std::string request_method, DirSettings setting);
```